src/shared/UI/button/Button.component.tsx

```diff
@@ -5,13 +5,13 @@ import { Spinner } from '../spinner/Spinner.component';
 
 import './Button.scss';
 
-interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
   buttonText?: string | ReactNode;
```

Comment:
buttonText can be ReactNode type ? Or rename buttonText varibale or delete ReactNode type

Reply:
buttonText can be an icon in this case. But I think that you are right. I need to refactor in with startIcon prop and endIcon prop.